library/std/src/thread/mod.rs

```diff
@@ -335,6 +342,16 @@ impl Builder {
         self
     }
 
+    /// Disables running and inheriting [spawn hooks](add_spawn_hook).
+    ///
+    /// Use this if the parent thread is in no way relevant for the child thread.
+    /// For example, when lazily spawning threads for a thread pool.
+    #[unstable(feature = "thread_spawn_hook", issue = "132951")]
+    pub fn no_hooks(mut self) -> Builder {
+        self.no_hooks = true;
+        self
+    }
```

Comment:
Should this be allowed when doing scoped thread spawning? Lazy spawning in a thread pool wouldn't use scoped thread spawning and denying `no_hooks` for scoped thread spawning could potentially allow `!Send` types to depend on thread local storage existing on all threads that can access it. For example if you have a C library which needs to be initialized on every thread, you may be able to pass a token around indicating that it is initialized and install a spawn hook to initialize it on every new thread. Then a `Sync + !Send` token would allow accessing the C api on child threads without having to check for initialization. This example may be a bit contrived though.